### PR TITLE
Replace WixSharp's MSI parser with a pure C# parser based on OpenMcdf

### LINKED ISSUE #79: [Bug]: PowerShell Deploy fails with "Value cannot be null. (Parameter 's')" (mac only)
### CommandLet

Deploy-WtWin32App

### Full command

Deploy-WtWin32App -PackageFolder "/Users/josef/Intune/Zoom.Zoom/6.0.39959" -Username info@example.com

### Module version

0.8.8

### What happened?

Some deploys fail, apparently because there is a parameter missing in the app.json.

It happened with the following packages:

Zoom.Zoom
KeePassXCTeam.KeePassXC
Google.Chrome

I used a pipe deploy: New-WtWingetPackage -PackageId "Zoom.Zoom" -PackageFolder "/Users/josef/Intune" | Deploy-WtWin32App -Username "info@example.com" 
(but it also happened with downloading it first and deploying it with a path to the folder)

### Command output

```PowerShell
INFO: [IntuneManager] Writing detection info with msi details Google.Chrome {829B647D-A3F1-39D5-B446-9D7EC9AB5705}
INFO: [IntuneManager] Writing package readme for package Google.Chrome
INFO: Loading package details from folder /Users/josef/Intune/Google.Chrome/126.0.6478.57
Deploy-WtWin32App: Value cannot be null. (Parameter 's')
```

### PowerShell version

7.4.2 (on a Mac)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I won't demand an immediate answer

### PR DESCRIPTION
**Pull request by miyoyo**
The current version of WinTuner uses WixSharp.UI.MsiParser to read the embedded database inside of msi files to get the product's code, and version, which is not available on 'nixes.

While the product code is easily available via winget, it appears that the product version field is not available, or not available _anymore_. Since the product version is required for msi deployment, and it is not available otherwise, this means msi packages cannot be created from 'nixes, like Linux or macOS.

This pull request implements a fallback to [msitools](https://wiki.gnome.org/msitools), a red hat/GNOME maintained project that implements support for msi file reading.

This package is available in all linux distros I have checked (including smaller ones like alpine and void) as well as macOS' [homebrew](https://formulae.brew.sh/formula/msitools).

I have run the Pester test suite on Linux, as well as manually tested New-WtWingetPackage, and it does manage to package msi files with no issues.

```
PS /mnt/c/Users/Hidden/Documents/WingetIntune> ./test.ps1

Starting discovery in 10 files.
Discovery found 28 tests in 467ms.
Running tests.
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/Deploy-WtMsStoreApp.Tests.ps1 419ms (65ms|202ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/Deploy-WtWin32App.Tests.ps1 57ms (2ms|25ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/Get-WtWin32Apps.Tests.ps1 43ms (2ms|12ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/New-IntuneWinPackage.Tests.ps1 38ms (2ms|8ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/NewWtWingetPackage.Tests.ps1 60ms (13ms|17ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/RemoveWtWin32App.Tests.ps1 54ms (4ms|22ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/Search-WtWinGetPackage.Tests.ps1 457ms (413ms|16ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/Unprotect-IntuneWinPackage.Tests.ps1 39ms (2ms|9ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/Update-WtIntuneApp.Tests.ps1 47ms (2ms|11ms)
[+] /mnt/c/Users/Hidden/Documents/WingetIntune/tests/WinTuner.Cmdlets.Tests/WinTuner.Tests.ps1 1.14s (1.03s|63ms)
Tests completed in 2.38s
Tests Passed: 28, Failed: 0, Skipped: 0, Inconclusive: 0, NotRun: 0
```

```
hidden@Prosopagnosia:/mnt/c/Users/Hidden/Documents/WingetIntune$ pwsh -Command "Import-Module ./dist/WinTuner/WinTuner.psd1; New-WtWingetPackage 7zip.7zip repo -TempFolder broken -Debug -Verbose; Remove-Module WinTuner"
INFO: Packaging package 7zip.7zip 24.09
INFO: [WingetManager] Getting package info for 7zip.7zip 24.09 from github
INFO: [ComputeBestInstallerForPackageCommand] Computing best installer for 7zip.7zip 24.09 Unknown
INFO: [IntuneManager] Generating IntuneWin package for 7zip.7zip 24.09 X64 System in repo
INFO: [IntuneManager] Downloading installer from https://7-zip.org/a/7z2409-x64.msi to broken/7zip.7zip/24.09/7z2409-x64.msi
INFO: [IntuneManager] Found 1 captures
INFO: [Packager] Creating Intune package from broken/7zip.7zip/24.09
INFO: [Packager] Creating package for broken/7zip.7zip/24.09/7z2409-x64.msi in broken/7zip.7zip/24.09 to repo/7zip.7zip/24.09
INFO: [Packager] Compressing the source folder broken/7zip.7zip/24.09 to /tmp/f99daf83-b0d8-4d63-8e8f-4c849a38c4ca/IntuneWinPackage/Contents/IntunePackage.intunewin
INFO: [Packager] Generating application info
INFO: [Packager] Encrypting file /tmp/f99daf83-b0d8-4d63-8e8f-4c849a38c4ca/IntuneWinPackage/Contents/IntunePackage.intunewin
INFO: [Packager] Generated detection XML file /tmp/f99daf83-b0d8-4d63-8e8f-4c849a38c4ca/IntuneWinPackage/Metadata/Detection.xml
INFO: [Packager] Done creating package for broken/7zip.7zip/24.09/7z2409-x64.msi in broken/7zip.7zip/24.09 to repo/7zip.7zip/24.09
INFO: [IntuneManager] Downloading logo from https://api.winstall.app/icons/7zip.7zip.png
INFO: [DefaultFileManager] Skipping download of https://api.winstall.app/icons/7zip.7zip.png to /mnt/c/Users/Hidden/Documents/WingetIntune/repo/7zip.7zip/logo.png because the file already exists
INFO: [IntuneManager] Writing detection info with msi details 7zip.7zip {23170F69-40C1-2702-2409-000001000000}
INFO: [IntuneManager] Writing package readme for package 7zip.7zip

PackageId          : 7zip.7zip
Version            : 24.09
PackageFolder      : repo/7zip.7zip/24.09
PackageFile        : 7z2409-x64.intunewin
InstallerFile      : 7z2409-x64.msi
InstallerArguments : -x64.msi /qn /norestart
```

Do note: this is the first C# code I've written in a very long time, so it may be better to cherry pick parts of this implementation if the code quality is insufficient, especially around error handling.

Closes #79

### CODE REVIEW

**Comment by svrooij** [comment]
I like you found a way to gather msi product codes on other platforms, but executing a process on the machine does not really provide a solution that also works in the cloud. If this is possible in an other language (so it seems) would it also be possible to do in c#?

**Comment by svrooij** [comment]
I spend a [great deal of time](https://svrooij.io/tags/intune) making this fully cross-platform, and want to remove **all** things that depend upon a binary that can only be run on Windows. That is why I'm quite hesitant against introducing another platform dependency.

I'm thinking Azure Functions, I know you can run those from a container and you could modify the built process to include another binary, but still that will cause a lot of headache. This is just a one-men project (for now), I'm not sure if I have the time to support this.

May I propose something else? What if you could specify a command format that would be executed to extract details from the msi.
`-MsiCommand "msitool.ps1 {msiLocation}"`, this command should then return `productCode;version`

Then there is no calling binaries inside this library and you are still able to extract msi details on linux. Other discussion is that this information should be correct in the Winget manifest....

**Comment by miyoyo** [comment]
That command argument would probably be fine, although it would change nothing (you would still require calling external binaries on Linux or it would break, and it would not need to do so on Windows thanks to WixSharp)

I'm trying to see if https://github.com/ironfede/openmcdf can be used to extract the version information, hopefully that can also remove WixSharp, as it's pure C#

**Comment by svrooij** [comment]
> That command argument would probably be fine, although it would change nothing (you would still require calling external binaries on Linux

This would change that I mark that as `Use this unsupported argument at your own risk` and then I would not have to deal with people that ask questions about that argument.

**Comment by miyoyo** [comment]
I definitely believe OpenMcdf is an option, I've already implemented decoding the string names and tables from an MSI, all I have to do now is figure out which string is the version and code string, but it's almost 2 AM, so that'll be tomorrow.

<details>

```cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Graph.Beta.Models;
using OpenMcdf;

namespace WingetIntune.Internal.Msi;
internal class MsiDecoder
{
    public string GetCode()
    {
        throw new NotImplementedException();
    }
    public string GetVersion()
    {
        throw new NotImplementedException();
    }
    public MsiDecoder(string filePath)
    {
        using (var cf = new CompoundFile(filePath))
        {
            var pool = LoadStringPool(cf);
        }
    }



    // references for the next lines:
    // https://stackoverflow.com/questions/9734978/view-msi-strings-in-binary

    private char BaseMSIDecode(char c)
    {
        // 0-0x3F converted to '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz._'
        // all other values higher as 0x3F converted also to '_'

        int result;

        if (c < 10)
            result = c + '0';             // 0-9 (0x0-0x9) -> '0123456789'
        else if (c < (10 + 26))
            result = c - 10 + 'A';        // 10-35 (0xA-0x23) -> 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
        else if (c < (10 + 26 + 26))
            result = c - 10 - 26 + 'a';   // 36-61 (0x24-0x3D) -> 'abcdefghijklmnopqrstuvwxyz'
        else if (c == (10 + 26 + 26))       // 62 (0x3E) -> '.'
            result = '.';
        else
            result = '_';                 // 63-0xffffffff (0x3F-0xFFFFFFFF) -> '_'

        return (char)result;
    }

    string DecodeStreamName(string name)
    {
        var result = new List<char>();
        var source = name.ToCharArray();

        foreach (char c in source)
        {
            var reduced = 0;
            if (c == 0x4840)
            {
                result.Add('$');
            }
            else if ((c >= 0x3800) && (c < 0x4840))
            {
                if (c >= 0x4800)
                {
                    reduced = c - 0x4800;
                    result.Add(BaseMSIDecode((char)(reduced)));
                }
                else
                {
                    reduced = c - 0x3800;
                    result.Add(BaseMSIDecode((char)(reduced & 0x3F)));
                    result.Add(BaseMSIDecode((char)((reduced >> 6) & 0x3F)));
                }
            }
            else
            {
                result.Add(c);
            }
        }

        return new string(result.ToArray());
    }

    private char BaseMSIEncode(char c)
    {
        // only '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz._' are allowed and converted to 0-0x3F

        int result;


        if ((c >= '0') && (c <= '9'))   // '0123456789' -> 0-9  (0x0-0x9)
            result = c - '0';
        else if ((c >= 'A') && (c <= 'Z'))   // 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' -> 10-35 (26 chars) - (0xA-0x23)
            result = c - 'A' + 10;
        else if ((c >= 'a') && (c <= 'z'))   // 'abcdefghijklmnopqrstuvwxyz' -> 36-61 (26 chars) - (0x24-0x3D)
            result = c - 'a' + 10 + 26;
        else if (c == '.')
            result = 10 + 26 + 26;        // '.' -> 62 (0x3E)
        else if (c == '_')
            result = 10 + 26 + 26 + 1;      // '_' -> 63 (0x3F) - 6 bits
        else
            result = -1; // other -> -1 (0xFF)
        return (char)result;
    }

    string EncodeStreamName(string name)
    {
        var result = new List<char>();

        for (int i = 0; i < name.Length; i++)
        {
            var c = name[i];
            if (c == '$')
            {
                result.Add((char)0x4840);
            }
            else if (c < 0x80 && BaseMSIEncode(c) <= 0x3F && i + 1 != name.Length)
            {
                i++;
                var first = BaseMSIEncode(c);
                var second = BaseMSIEncode(name[i]);
                result.Add((char)(first + (second << 6) + 0x3800));
            }
            else
            {
                result.Add((char)(BaseMSIEncode(c) + 0x4800));
            }

        }

        return new string(result.ToArray());
    }

    Dictionary<int, string> LoadStringPool(CompoundFile cf)
    {
        var decodedStringPool = EncodeStreamName("$_StringPool");
        var streamStringPool = cf.RootStorage.GetStream(decodedStringPool);
        var stringPoolBytes = streamStringPool.GetData();
        var poolLength = streamStringPool.Size;
        var poolWLength = BitConverter.ToInt16(stringPoolBytes, 0);
        var poolRefCount = BitConverter.ToInt16(stringPoolBytes, 2);

        var decodedStringData = EncodeStreamName("$_StringData");
        var streamStringData = cf.RootStorage.GetStream(decodedStringData);
        var stringDataBytes = streamStringData.GetData();

        var strings = new Dictionary<int, string>();

        for (int src = 4, stringId = 1, offset = 0; src < poolLength; src += 4)
        {
            Console.WriteLine("Starting decode");
            var entryLength = (int)BitConverter.ToInt16(stringPoolBytes, src);
            var entryRef = (int)BitConverter.ToInt16(stringPoolBytes, src + 2);

            Console.WriteLine($"Of entry {entryLength} {entryRef}");

            if (entryLength == 0 && entryRef == 0)
            {
                // Empty entry, skip.
                Console.WriteLine("Skipping");
                stringId++;
                continue;
            }
            else if (entryLength == 0 && entryRef != 0)
            {
                // wide entry over 64kb
                Console.WriteLine("Wide Entry");
                continue;
            }

            if (src != 4)
            {

                var previousEntryLength = BitConverter.ToInt16(stringPoolBytes, src - 4);
                var previousEntryRef = BitConverter.ToInt16(stringPoolBytes, src - 2);
                Console.WriteLine($"Previous entry {previousEntryLength} {previousEntryRef}");

                if (previousEntryLength == 0 && previousEntryRef != 0)
                {
                    entryLength += previousEntryLength << 16;
                    Console.WriteLine($"New Size {entryLength}");
                }
            }

            Console.WriteLine($"Adding {Encoding.UTF8.GetString(stringDataBytes.Skip(offset).Take(entryLength).ToArray())}");

            strings.Add(stringId, Encoding.UTF8.GetString(stringDataBytes.Skip(offset).Take(entryLength).ToArray()));
            offset += entryLength;
            stringId++;
        }

        return strings;
    }
}

```

</details>

**Comment by miyoyo** [comment]
Welp, looks like addiction won in the end, it works.

<details>

```cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Graph.Beta.Models;
using OpenMcdf;

namespace WingetIntune.Internal.Msi;
internal class MsiDecoder
{
    private int stringSize = 2;
    private Dictionary<int, string> intToString;
    private Dictionary<string, int> stringToInt;
    public string GetCode()
    {
        throw new NotImplementedException();
    }
    public string GetVersion()
    {
        throw new NotImplementedException();
    }
    public MsiDecoder(string filePath)
    {
        using (var cf = new CompoundFile(filePath))
        {
            intToString = LoadStringPool(cf);
            stringToInt = intToString.ToDictionary(x => x.Value, x => x.Key);

            foreach(var entry in intToString)
            {
                Console.WriteLine($"{entry.Key}: {entry.Value}");
            }

            var decodedPropertyName = EncodeStreamName("$Property");
            var streamProperty = cf.RootStorage.GetStream(decodedPropertyName);
            var propertyBytes = streamProperty.GetData();
            var enubytes = Enumerable.Range(0, propertyBytes.Length / 2).Select(i => BitConverter.ToUInt16(propertyBytes, i * 2)).ToArray();
            var cells = new List<string>();

            foreach (var b in enubytes)
            {
                cells.Add(intToString[b]);
            }

            var tableSize = cells.Count() / 2;

            var code = cells[cells.IndexOf("ProductCode") + tableSize];
            var version = cells[cells.IndexOf("ProductVersion") + tableSize];

            Console.WriteLine($"Code: {code}, Version: {version}");
        }
    }



    // references for the next lines:
    // https://stackoverflow.com/questions/9734978/view-msi-strings-in-binary

    private char BaseMSIDecode(char c)
    {
        // 0-0x3F converted to '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz._'
        // all other values higher as 0x3F converted also to '_'

        int result;

        if (c < 10)
            result = c + '0';             // 0-9 (0x0-0x9) -> '0123456789'
        else if (c < (10 + 26))
            result = c - 10 + 'A';        // 10-35 (0xA-0x23) -> 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
        else if (c < (10 + 26 + 26))
            result = c - 10 - 26 + 'a';   // 36-61 (0x24-0x3D) -> 'abcdefghijklmnopqrstuvwxyz'
        else if (c == (10 + 26 + 26))       // 62 (0x3E) -> '.'
            result = '.';
        else
            result = '_';                 // 63-0xffffffff (0x3F-0xFFFFFFFF) -> '_'

        return (char)result;
    }

    string DecodeStreamName(string name)
    {
        var result = new List<char>();
        var source = name.ToCharArray();

        foreach (char c in source)
        {
            var reduced = 0;
            if (c == 0x4840)
            {
                result.Add('$');
            }
            else if ((c >= 0x3800) && (c < 0x4840))
            {
                if (c >= 0x4800)
                {
                    reduced = c - 0x4800;
                    result.Add(BaseMSIDecode((char)(reduced)));
                }
                else
                {
                    reduced = c - 0x3800;
                    result.Add(BaseMSIDecode((char)(reduced & 0x3F)));
                    result.Add(BaseMSIDecode((char)((reduced >> 6) & 0x3F)));
                }
            }
            else
            {
                result.Add(c);
            }
        }

        return new string(result.ToArray());
    }

    private char BaseMSIEncode(char c)
    {
        // only '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz._' are allowed and converted to 0-0x3F

        int result;


        if ((c >= '0') && (c <= '9'))   // '0123456789' -> 0-9  (0x0-0x9)
            result = c - '0';
        else if ((c >= 'A') && (c <= 'Z'))   // 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' -> 10-35 (26 chars) - (0xA-0x23)
            result = c - 'A' + 10;
        else if ((c >= 'a') && (c <= 'z'))   // 'abcdefghijklmnopqrstuvwxyz' -> 36-61 (26 chars) - (0x24-0x3D)
            result = c - 'a' + 10 + 26;
        else if (c == '.')
            result = 10 + 26 + 26;        // '.' -> 62 (0x3E)
        else if (c == '_')
            result = 10 + 26 + 26 + 1;      // '_' -> 63 (0x3F) - 6 bits
        else
            result = -1; // other -> -1 (0xFF)
        return (char)result;
    }

    string EncodeStreamName(string name)
    {
        var result = new List<char>();

        for (int i = 0; i < name.Length; i++)
        {
            var c = name[i];
            if (c == '$')
            {
                result.Add((char)0x4840);
            }
            else if (c < 0x80 && BaseMSIEncode(c) <= 0x3F && i + 1 != name.Length)
            {
                i++;
                var first = BaseMSIEncode(c);
                var second = BaseMSIEncode(name[i]);
                result.Add((char)(first + (second << 6) + 0x3800));
            }
            else
            {
                result.Add((char)(BaseMSIEncode(c) + 0x4800));
            }

        }

        return new string(result.ToArray());
    }

    Dictionary<int, string> LoadStringPool(CompoundFile cf)
    {
        var decodedStringPool = EncodeStreamName("$_StringPool");
        var streamStringPool = cf.RootStorage.GetStream(decodedStringPool);
        var stringPoolBytes = streamStringPool.GetData();
        var poolLength = streamStringPool.Size;
        var poolWLength = BitConverter.ToUInt16(stringPoolBytes, 0);
        var poolRefCount = BitConverter.ToUInt16(stringPoolBytes, 2);

        if (poolRefCount == 0)
            stringSize = 2;
        else if (poolRefCount == 0x8000)
            stringSize = 3;

        var decodedStringData = EncodeStreamName("$_StringData");
        var streamStringData = cf.RootStorage.GetStream(decodedStringData);
        var stringDataBytes = streamStringData.GetData();

        var strings = new Dictionary<int, string>();

        for (int src = 4, stringId = 1, offset = 0; src < poolLength; src += 4)
        {
            Console.WriteLine("Starting decode");
            var entryLength = (int)BitConverter.ToUInt16(stringPoolBytes, src);
            var entryRef = (int)BitConverter.ToUInt16(stringPoolBytes, src + 2);

            Console.WriteLine($"Of entry {entryLength} {entryRef}");

            if (entryLength == 0 && entryRef == 0)
            {
                // Empty entry, skip.
                Console.WriteLine("Skipping");
                stringId++;
                continue;
            }
            else if (entryLength == 0 && entryRef != 0)
            {
                // wide entry over 64kb
                Console.WriteLine("Wide Entry");
                continue;
            }

            if (src != 4)
            {

                var previousEntryLength = BitConverter.ToInt16(stringPoolBytes, src - 4);
                var previousEntryRef = BitConverter.ToInt16(stringPoolBytes, src - 2);
                Console.WriteLine($"Previous entry {previousEntryLength} {previousEntryRef}");

                if (previousEntryLength == 0 && previousEntryRef != 0)
                {
                    entryLength += previousEntryLength << 16;
                    Console.WriteLine($"New Size {entryLength}");
                }
            }

            Console.WriteLine($"Adding {Encoding.UTF8.GetString(stringDataBytes.Skip(offset).Take(entryLength).ToArray())}");

            strings.Add(stringId, Encoding.UTF8.GetString(stringDataBytes.Skip(offset).Take(entryLength).ToArray()));
            offset += entryLength;
            stringId++;
        }

        return strings;
    }
}
```

</details>

**Comment by miyoyo** [comment]
Alright, pushed my changes, this version is not perfect yet, as I'm parsing large numbers (i4) incorrectly, otherwise, parsing numbers is sufficient, and adding queries can be done in a similar way as both GetCode and GetVersion.

Probably deserves a cleanup before it's accepted too.

**Comment by svrooij** [comment]
If this all works as expected, I'm also for removing the wixsharp code that is included here: https://github.com/svrooij/WingetIntune/tree/main/src/WingetIntune/Internal/Msi

The idea is that it will use this implementation on all platforms right?

**Comment by miyoyo** [comment]
> If this all works as expected, I'm also for removing the wixsharp code that is included here: https://github.com/svrooij/WingetIntune/tree/main/src/WingetIntune/Internal/Msi
> 
> The idea is that it will use this implementation on all platforms right?

Yeah, since this is pure C#, I don't see a reason to keep WixSharp around

**Comment by svrooij** [comment]
I think it might be a good idea to add a test that uses this new code.
But for the rest I think it looks good

**Comment by miyoyo** [comment]
Alright, do you think It's better to embed an msi file, or redownload one for the test?

I'll write the test(s) tonight (gmt+1) otherwise.

**Comment by svrooij** [comment]
If you can find a really small MSI somewhere I would add that to the tests folder and use that. Otherwise just take an url from somewhere.

**Comment by miyoyo** [comment]
There we go, I'm using a small msi from Microsoft's servers, so I'm reasonably confident it'll stay in place for a while.

I'm only testing the publicly exposed interface, since that's what's in use, and I don't see the point in testing it from the IntuneManager side, since it'd basically be the same code, just with more effort, as OpenMcdf reads the files itself otherwise.

**Comment by miyoyo** [comment]
No clue why the tests failed, there shouldn't be an impact on them, but I just imported MSAL directly.

Also ran formatter (dammit I keep forgetting that one)

**Comment by svrooij** [comment]
That did not go as plannend, let me try again
